doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,231 @@
+
+
+Zephyr CMake package export (west)
+**********************************
+
+When installing Zephyr using :ref:`west <get_the_code>` then it is recommended to export Zephyr
+using ``west zephyr-export``.
+
+
+Zephyr CMake package export (without west)
+******************************************
+
+Zephyr can be exported using CMake directly.
+To do so, run the following command:
+
+``cmake -S <PATH-TO-ZEPHYR>/share/zephyr-package/cmake -B <PATH-TO-ZEPHYR>/share/zephyr-package/cmake``
+
+
+this will export the current Zephyr to CMake user package registry.
+
+.. _zephyr_cmake_package_zephyr_base:
+
+Zephyr Base Environment Setting
+*******************************
+
+The Zephyr CMake package search functionality allows for explicitly specifying a Zephyr base using
+an environment variable.
+
+In order to support the use of ``ZEPHYR_BASE`` environment setting, the following ``find_package()``
+syntax much be used:
+
+  .. code-block:: cmake
+
+    find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
+
+This syntax instructs CMake to first search for Zephyr at the location pointed to by ``$ZEPHYR_BASE``
+and if that is unset, then use the normal search pattern.
+
+.. _zephyr_cmake_search_order:
+
+Zephyr CMake Package Search Order
+*********************************
+
+When ``ZEPHYR_BASE`` is not used for searching, then the Zephyr matching the following criteria will
+be used
+
+* Zephyr in-tree: This means a sample located inside an existing tree, will automatically use the Zephyr in same tree.
+  As example::
+
+      <path-to>/zephyr-worktree/zephyr
+      <path-to>/zephyr-worktree/zephyr/samples/hello_world
+
+  in this example, ``hello_world`` will use ``<path-to>/zephyr-worktree/zephyr`` cause the Zephyr
+  and the application shares the same tree.
+
+
+* Zephyr in-working-tree: This means a sample located in a project tree, will use the Zephyr found in same work-tree.
+  As example::
+
+      <path-to>/other-worktree/zephyr
+
+      <path-to>/zephyr-worktree/zephyr
+      <path-to>/zephyr-worktree/applications/hello_world
+
+  in this example, ``hello_world`` will use ``<path-to>/zephyr-worktree/zephyr`` as they share the
+  same work-tree.
+
+
+* Out-of-tree: This means a sample located out-side any Zephyr tree.
+  In this case, the application will use the first Zephyr from CMake user package registry.
+  As example::
+
+      <path-to>/other-worktree/zephyr  (not exported to CMake)
+
+      <path-to>/zephyr-worktree/zephyr (exported to CMake)
+
+      <path-to>/applications/hello_world
+
+  in this example, only ``<path-to>/zephyr-worktree/zephyr`` is exported to the CMake package
+  registry and therefore this Zephyr will be used.
+
+  If user wants to test the application with ``<path-to>/other-worktree/zephyr``, this can be done
+  by setting ``ZEPHYR_BASE=<path-to>/other-worktree/zephyr`` in environment, before running CMake.
+
+
+Zephyr CMake Package Version (Out-of-tree)
```

Comment:
I think "out of tree" might be confusing here. It might mean one of two things:

1. application outside of the main zephyr repository
2. application outside of a west workspace

Can this be clarified by using the term "freestanding Zephyr application" or something like that? It's fine if you're happy with this as is.

Reply:
i'm not happy, would like something better, but it was the best I could find, so far.

Reply:
Updated according to discussion